### PR TITLE
LPS-160210 Add WebDAV URL to Info Panel

### PR DESCRIPTION
**Pull request by cgoncas**
[Jira ticket](https://issues.liferay.com/browse/LPS-160210)

## Motivation

Add WebDAV URL to the info panel in Content Dashboard.

## Proposed Solution

Send URL fields in the SpecificFields section of the infoPanel endpoint:

```
"specificFields": {
    "extension": {
      "title": "Extension",
      "type": "String",
      "value": "png"
    },
    "size": {
      "title": "Size",
      "type": "String",
      "value": "359 KB"
    },
    "web-dav-url": {
      "title": "WebDAV URL",
      "type": "URL",
      "value": "http://localhost:8080/webdav/guest/document_library/tree.png"
    },
    "latest-version-url": {
      "title": "Latest Version URL",
      "type": "URL",
      "value": "http://localhost:8080/documents/d/guest/tree-png"
    }
  }
```

Support URL fields in the Content Dashboard Info Panel Section

## Screencaps

image

## How to

- Upload some files to the DL
- Navigate to the CD
- Open the info panel
- Check for the components in the details section

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

I can't sort it alphabetically, because once I print it in JSON, the order changes again :(

Thanks!!!

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

As you mentioned, there are some JS test failing in this PR :(.

Thanks!!

**Comment by cgoncas** [comment]
> As you mentioned, there are some JS test failing in this PR :(.

Hey @beltranrengifo !!

I have updated the Snapshot, wish us luck :)

Thanks!!

**Comment by cgoncas** [comment]
ci:test:relevant

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/SidebarPanelInfoView.js [hunk 1]
```diff
@@ -91,7 +91,17 @@ const SidebarPanelInfoView = ({
...
+	const specificItems = Object.values(specificFields);
+
+	const renderSpecificItem = ({languageTag, type, value}) => {
+		const options = {
+			Date: () => formatDate(value, languageTag),
+			URL: () => <FileUrlCopyButton url={value} />,
+			default: () => value,
+		};
+
+		return (options[type] && options[type]()) || options.default();
```
Comment (beltranrengifo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/SidebarPanelInfoView.js:103]: I've reinvented the ternary here 🤣 
I should change it later

**Comment by beltranrengifo** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**